src/BSON/Javascript.c

```diff
@@ -215,6 +216,41 @@ PHP_METHOD(Javascript, getScope)
 }
 /* }}} */
 
+/* {{{ proto array Javascript::jsonSerialize()
+*/
+PHP_METHOD(Javascript, jsonSerialize)
+{
+	php_phongo_javascript_t *intern;
+	bool has_scope;
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		return;
+	}
+
+	intern = Z_JAVASCRIPT_OBJ_P(getThis());
+
+	has_scope = (intern->scope && intern->scope->len);
+	array_init_size(return_value, has_scope ? 2 : 1);
+	ADD_ASSOC_STRINGL(return_value, "$code", intern->code, intern->code_len);
```

Comment:
Waiting on a response from the C driver team. I imagine they may keep their non-compliant output format (i.e. code as a string and ignoring scope) for BC until 2.0 but allow the `$code`/`$scope` format to be parsed properly when they address the ticket in 1.6.0.

I'm all in favor of sticking to the spec and returning a compliant value today, even if libbson cannot parse it yet.